src/daft-local-execution/src/sources/scan_task.rs

```diff
@@ -241,19 +241,14 @@ async fn stream_scan_task(
     }
     let source = scan_task.sources.first().unwrap();
     let url = source.get_path();
-    let (io_config, multi_threaded_io) = match scan_task.storage_config.as_ref() {
-        StorageConfig::Native(native_storage_config) => (
-            native_storage_config.io_config.as_ref(),
-            native_storage_config.multithreaded_io,
-        ),
-
-        #[cfg(feature = "python")]
-        StorageConfig::Python(python_storage_config) => {
-            (python_storage_config.io_config.as_ref(), true)
-        }
-    };
-    let io_config = Arc::new(io_config.cloned().unwrap_or_default());
-    let io_client = daft_io::get_io_client(multi_threaded_io, io_config)?;
+    let io_config = Arc::new(
+        scan_task
+            .storage_config
+            .io_config
+            .clone()
+            .unwrap_or_default(),
+    );
```

Comment:
This feels a little odd -- should we have `StorageConfig` hold an `Arc<IOConfig>` instead? I'm guessing the same IOConfig is going to be passed around a lot.

Reply:
Makes sense, will change that!

Reply:
Actually this would probably make sense as a larger change since we would probably want to also make things like the Python IOConfig as well as DaftPlanningConfig also store an arc. Since this is not a regression, I think I will also defer this to a future PR.

Reply:
Yes that makes sense, let's defer